bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/SmartHomeDeviceHandler.java

```diff
@@ -86,95 +157,127 @@ public void initialize() {
+                                        new PercentType(value.get("brightness").getAsInt() * 100));
+                            }
+                        }
+                    } else if ("Alexa.ColorPropertiesController".equals(interfaceName)) {
+                        Set<String> capabilities = GetCapabilities(accountHandler, smartHomeBaseDevice);
+                        if ("colorProperties".equals(state.get("name").getAsString())) {
+                            if (capabilities.contains(INTERFACE_COLOR_TEMPERATURE)) {
+                                // For groups take the first
+                                if (colorTemperaturName == null) {
+                                    colorTemperaturName = state.get("value").getAsJsonObject().get("name")
+                                            .getAsString();
                                 }
-                                if (color != null) {
-                                    updateColor(thing.getChannel(CHANNEL_LIGHT_COLOR).getUID(), color);
+                            } else if (capabilities.contains(INTERFACE_COLOR)) {
+                                // For groups take the first
+                                if (colorName == null) {
+                                    colorName = state.get("value").getAsJsonObject().get("name").getAsString();
                                 }
-                            } catch (IOException | URISyntaxException e) {
-                                logger.error(e.getMessage());
+                            }
+                        }
+                    } else if (INTERFACE_SECURITY_PANEL.equals(interfaceName)) {
+                        if ("armState".equals(state.get("name").getAsString())) {
+                            // For groups take the first
+                            if (armState == null) {
+                                armState = state.get("value").getAsString();
                             }
                         }
                     }
                 }
-            };
-
-            Configuration config = accountHandler.getThing().getConfiguration();
-
-            updateStateJob = scheduler.scheduleWithFixedDelay(runnable, 0,
-                    ((BigDecimal) config.getProperties().get("pollingIntervalSmartHome")).longValue(),
-                    TimeUnit.SECONDS);
-        }
-    }
-
-    public void addChannelToDevice(ChannelUID channelUID, String itemType, ChannelTypeUID channelTypeUID) {
-        updateThing(editThing()
-                .withChannel(ChannelBuilder.create(channelUID, itemType).withType(channelTypeUID).build()).build());
-    }
-
-    @Override
-    public void dispose() {
-        ScheduledFuture<?> updateStateJob = this.updateStateJob;
-        this.updateStateJob = null;
-        if (updateStateJob != null) {
-            updateStateJob.cancel(false);
+            }
+            if (this.isLinked(CHANNEL_CONTROLLER_POWER_STATE)) {
```

Comment:
**We should discuss the handling of channels in groups**. Currently I used your implemention, the Switch takes ON if one item is on, the brightness takes the maximum. But all other items uses the first value. Maybe it would be better to use UNDEF if the values are different.